### PR TITLE
some fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/upnp/quirks.cc [hunk 1]
```diff
@@ -55,7 +55,7 @@ long Quirks::checkFlags(long flags) const
 
 bool Quirks::hasFlag(long flag) const
 {
-    return pClientInfo && (pClientInfo->flags & flag) == flag;
+    return pClientInfo && (static_cast<long>(pClientInfo->flags) & flag) == flag;
```
Comment (KarlStraussberger) [inline src/upnp/quirks.cc:58]: There is another such line in `checkFlags`. But I prefer to replace the long by `QuirkFlags`, which requires changes in `ConfigOption` as well.
As I'm already in the middle of these changes, you can remove that change.

File: src/util/mt_inotify.cc [hunk 2]
```diff
@@ -127,7 +127,7 @@ struct inotify_event* Inotify::nextEvent()
     // first_byte is index into event buffer
     if (firstByte != 0
         && firstByte <= int(bytes - sizeof(struct inotify_event))) {
-        ret = reinterpret_cast<struct inotify_event*>(reinterpret_cast<char*>(event.data()) + firstByte);
+        std::memcpy(ret, reinterpret_cast<char*>(event.data()) + firstByte, sizeof(inotify_event));
```
Comment (KarlStraussberger) [inline src/util/mt_inotify.cc:130]: There is another `copy_n` in the code below. We should only use one of `copy_n` or `memcpy`

File: src/metadata/taglib_handler.cc [hunk 3]
```diff
@@ -419,7 +419,7 @@ std::unique_ptr<IOHandler> TagLibHandler::serveContent(const std::shared_ptr<Cds
         if (!f || !f->isValid() || !f->tag())
             throw_std_runtime_error("Could not open file {} as OGG", itemLocation);
 
-        const TagLib::List<TagLib::FLAC::Picture*> picList = reinterpret_cast<TagLib::Ogg::XiphComment*>(f->tag())->pictureList();
+        const auto picList = static_cast<TagLib::Ogg::XiphComment*>(f->tag())->pictureList();
```
Comment (KarlStraussberger) [inline src/metadata/taglib_handler.cc:422]: From the spec `static_cast` produces extra code to align the pointers. Do you see the need for this.

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

File: src/upnp/quirks.cc (hunk 1 above)
Comment (neheb) [reply src/upnp/quirks.cc]: Removed.

File: src/util/mt_inotify.cc (hunk 2 above)
Comment (neheb) [reply src/util/mt_inotify.cc:130]: Probably the latter since it's faster (copy_n uses memmove).

File: src/metadata/taglib_handler.cc (hunk 3 above)
Comment (neheb) [reply src/metadata/taglib_handler.cc]: Removed.

Comment (neheb) [reply src/metadata/taglib_handler.cc]: So clang-tidy actually warns on this: https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-static-cast-downcast.html

looks like dynamic_cast is the proper thing to use here.

**Pull request merged**